File: tests/payloads/suite/java/java/sieve.java
import java.util.Scanner;
import java.util.Arrays;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt()) return;
        int n = sc.nextInt();
        
        if (n < 2) {
            System.out.println(0);
            return;
        }

        byte[] s = new byte[n + 1];
        Arrays.fill(s, (byte) 1);
        s[0] = 0;
        s[1] = 0;

        int limit = (int) Math.sqrt(n);
        for (int i = 2; i <= limit; i++) {
            if (s[i] == 1) {
                for (int j = i * i; j <= n; j += i) {
                    s[j] = 0;
                }
            }
        }

        long count = 0;
        for (int i = 0; i <= n; i++) {
            if (s[i] == 1) {
                count++;
            }
        }
        System.out.println(count);
    }
}

File: tests/payloads/suite/cpp/sieve.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int n;
    if (!(cin >> n)) return 0;
    if (n < 2) {
        cout << 0 << endl;
        return 0;
    }
    vector<char> s(n + 1, 1);
    s[0] = s[1] = 0;
    int limit = sqrt(n);
    for (int i = 2; i <= limit; ++i) {
        if (s[i]) {
            for (int j = i * i; j <= n; j += i) {
                s[j] = 0;
            }
        }
    }
    long long count = 0;
    for (int i = 0; i <= n; ++i) {
        if (s[i]) count++;
    }
    cout << count << endl;
    return 0;
}
